AmericanMedicalAssociation/ama-style-guide-2
# EWL-10393: Hub page banner styling.

## Ticket(s)
**Do not** submit a Pull Request without a relevant JIRA ticket or Github issue! If you are creating a new pattern or feature, create an issue describing the need for this feature in Github **first** and then link to it below.

**Github Issue**
- [Issue XXX: Issue Title](https://github.com/AmericanMedicalAssociation/AMA-style-guide/issues/XXX)

## JIRA Ticket(s)
- [EWL-10393: Hub Page | Sticky Banner Styling](https://issues.ama-assn.org/browse/EWL-10393)


## Description:
Add styling for styling hub page banner and sticky functionality


## To Test:
- checkout ama-d8 PR: https://github.com/AmericanMedicalAssociation/ama-d8/pull/4290
- clear caches
- create Hub Page Banner custom block
- edit any Hub page and add newly built custom block to the Hub Page Banner field
- confirm styling of banner matches the following zeplin: https://app.zeplin.io/project/64df68b2df6fee220c92edcf/screen/64df68eddf6fee220c92ef07
- confirm banner stays attached to the bottom of the page until you scroll to the footer, at which point the banner should stop and rest on top of the footer.

- run styleguide (gulp serve)
- navigate to Hub > Hub Baner
- confirm styling matches the following zeplins: https://app.zeplin.io/project/64df68b2df6fee220c92edcf/screen/64df68eddf6fee220c92ef07
https://app.zeplin.io/project/64df68b2df6fee220c92edcf/screen/65033af4f77df067d0e3dbfb

## Visual Regressions

_Please provide the pull request ID below. This will then reference the automated VRT report after it has been generated._

A report is available [here](http://htmlpreview.github.io/?https://github.com/AmericanMedicalAssociation/ama-style-guide-2/blob/visual-regression-testing-artifact/[PULL_REQUEST_ID]/html_report/index.html).

**Before proceeding:** Run visual regression tests locally to ensure new work does not introduce new bugs.

_The visual regression tests always compare against the production version of the style guide. Changes and new work will show up as test failures. Please describe your work so that the report can be reviewed by the team._

If you are creating or updating a pattern be sure you have created or updated the [scenario in `backstop.json`](ama-style-guide-2/styleguide/backstop.json).

_After changes in the report are documented and demonstrate your desired changes, please mark the pull request as `ready for review`.

For more information on visual regression testing review the [How do I run tests?](https://issues.ama-assn.org:8446/confluence/pages/viewpage.action?pageId=23298568) document in Confluence.


## Relevant Screenshots/GIFs
Use something like [Skitch](https://evernote.com/skitch/) or [GIPHY Capture](https://giphy.com/apps/giphycapture) to capture images/gifs to demonstrate behaviors.


## Remaining Tasks
Things relevant to here may be updating [The Glossary](https://issues.ama-assn.org:8446/confluence/display/DTD/Glossary?src=contextnavpagetreemode) with the latest naming changes or review of the relative [D8](https://github.com/AmericanMedicalAssociation/ama-d8) work.


## Additional Notes
Anything more to add? Good things to call out here are:
- are there any PRs in this or other repositories that relate to or affect this PR?
- are there any caveats or oddities testers should know about when testing this PR?
- are there any problems you ran into during your work that you'd like to call out?

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)
